### PR TITLE
How to send SMS in Django using Twilio

### PR DESCRIPTION
**Pull request by shosenwales**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [yes ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by shosenwales** [comment]
This fixes #How to send SMS in Django using Twilio

**Comment by shosenwales** [comment]
This fixes #2098

**Comment by shosenwales** [comment]
Thanks @geoffrey45

File: articles/how-to-send-sms-in-django-using-twilio/index.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+Here, we specify the `/path/to/new/virtual/environment` as `env`.
+
+On creation, you can activate the virtual environment using the following command:
+
+```bash
+$ source env/bin/activate
+```
+
+On activating the environment, we can install Django using the following command:
```
Comment (geoffrey45) [inline articles/how-to-send-sms-in-django-using-twilio/index.md:42]: Don't mix pronouns. Choose one and stick to it throughout your tutorial. Ie. Choose between `you` and `we` and use it throughout.

File: articles/how-to-send-sms-in-django-using-twilio/index.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+```python
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'score',
+    ]
```
Comment (geoffrey45) [inline articles/how-to-send-sms-in-django-using-twilio/index.md:71]: Use the correct indentation here.

File: articles/how-to-send-sms-in-django-using-twilio/index.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+```bash
+# migrating the app and database changes
+$ python manage.py makemigrations
+
+# final migrations
+$ python manage.py migrate
+```
+
+A superuser has the permissions to create, edit, update, and delete data in Django admin. We create a superuser by running the command below:
+
```
Comment (geoffrey45) [inline articles/how-to-send-sms-in-django-using-twilio/index.md:123]: Write the second sentence as an instruction for flow. Ie. `Create a superuser by ...`

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@shosenwales Thanks for submitting this amazing tutorial. It's straight to the point. 👏

However, there are some minor changes I've suggested for flow. Please see the comments attached below.

**Comment by shosenwales** [comment]
Hi @geoffrey45  
I've fixed the listed issues. kindly go through

Thanks.

**Comment by geoffrey45** [comment]
Hello  @shosenwales   :wave:

As you know, we were working on a new structure for our EngEd Repo, which led to a slight slow down of article publishing. Everything is almost ready now.

In order to publish your article as soon as possible, please do the following:
- Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
- Delete your article directory .ie. `how-to-send-sms-in-django-using-twilio` from the `articles` folders.
- Do a git pull (take a look at the new structure of the repo after a git pull).
- Upload your article files to the [new *articles* folder inside *content*](https://github.com/section-engineering-education/engineering-education/tree/master/content/articles) ie. `/content/articles`.

Ping me via Slack if you have any problems.

**Review by geoffrey45 — APPROVED** [review]

**Comment by hectorkambow** [comment]
Thank you v much for your patience as we worked through our Repo re-structuring process. 

The purpose behind this restructure - was to better allow us to handle more volume and allow students to publish more articles and get more involved with the community (with open-source contribution and suggestions). 

With most of the changes coming to completion - we will be sure to get your article published asap. 

Thank you again - and we are looking forward to more amazing articles.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
great work Gents @shosenwales 

awesome tutorial - very easy to follow along 👍 

cc @geoffrey45

**Pull request merged**